### ISSUE TITLE
spinel: external headers should not be included by core spinel protocol

### ISSUE THREAD

**Issue by turon** [openthread/openthread#121]
Embedded platforms may not have access to this header or the in6_addr structure included by spinel.h:

[spinel.h]
```
 #include <arpa/inet.h>
// ...
typedef struct in6_addr spinel_ipv6addr_t;
```

Perhaps moving over to using openthread-types.h would be more appropriate?

```
#include <openthread-types.h>
// ...
typedef otIp6Address spinel_ipv6addr_t;
```

**Comment by turon** [comment]
Here is a somewhat related issue where ncp_base.cpp assumes being built by autotools, thus making the NCP code not portable across toolchains:

[ncp_base.cpp]
```
#include <openthread-config.h>
// ...
// line 908
        PACKAGE_NAME "/" PACKAGE_VERSION "; " __DATE__ " " __TIME__
```

**Comment by darconeous** [comment]
Good catch. Will fix.

**Comment by darconeous** [comment]
Will likely try to avoid using OpenThread types, so that the file can be used in other projects easily without depending on the rest of OpenThread.

**Comment by jwhui** [comment]
Fixed in #133.
